components/support/sync15-traits/src/payload.rs

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use super::Guid;
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value as JsonValue};
+
+/// Represents the decrypted payload in a Bso. Provides a minimal layer of type
+/// safety to avoid double-encrypting.
+///
+/// Note: If we implement a full sync client in rust we may want to consider
```

Comment:
I guess we did end up implementing a full client, after all, and this abstraction seemed to work out pretty well. 😆 Since you're here, anyway (I wish Git supported renaming files, not just remove + add), mind nixing this comment?